sagemaker-clarify/online_explainability/tabular_multi_model_endpoint/multi_model_xgboost_with_online_explainability.ipynb

```diff
@@ -0,0 +1,1823 @@
+    "    \"\"\"\n",
+    "    Visualize the given SHAP values with an additive force layout.\n",
+    "\n",
+    "    For more information: https://shap.readthedocs.io/en/latest/example_notebooks/tabular_examples/tree_based_models/Force%20Plot%20Colors.html\n",
+    "    \"\"\"\n",
+    "    force_plot_display = shap.plots.force(\n",
+    "        base_value=expected_value,\n",
+    "        shap_values=shap_values,\n",
+    "        features=feature_data,\n",
+    "        feature_names=feature_headers,\n",
+    "        matplotlib=True,\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "def display_plots(explanations, expected_value, request_records, predictions):\n",
+    "    \"\"\"\n",
+    "    Display the Model Explainability plots\n",
+    "    \"\"\"\n",
+    "    per_request_shap_values = OrderedDict()\n",
+    "    feature_headers = []\n",
+    "    for i, record_output in enumerate(explanations):\n",
+    "        per_record_shap_values = []\n",
+    "        if record_output is not None:\n",
+    "            feature_headers = []\n",
+    "            for feature_attribution in record_output:\n",
+    "                per_record_shap_values.append(\n",
+    "                    feature_attribution[\"attributions\"][0][\"attribution\"][0]\n",
+    "                )\n",
+    "                feature_headers.append(feature_attribution[\"feature_header\"])\n",
+    "            per_request_shap_values[i] = per_record_shap_values\n",
+    "\n",
+    "    for record_index, shap_values in per_request_shap_values.items():\n",
+    "        print(\n",
+    "            f\"Visualize the SHAP values for Record number {record_index + 1} with Model Prediction: {predictions[0][record_index]}\"\n",
+    "        )\n",
+    "        force_plot(\n",
+    "            expected_value,\n",
+    "            np.array(shap_values),\n",
+    "            request_records.iloc[record_index],\n",
+    "            feature_headers,\n",
+    "        )\n",
+    "\n",
+    "\n",
+    "def visualize_result(result, request_records, expected_value):\n",
+    "    \"\"\"\n",
+    "    Visualize the output from the endpoint.\n",
+    "    \"\"\"\n",
+    "    predictions = pd.read_csv(io.StringIO(result[\"predictions\"][\"data\"]), header=None)\n",
+    "    predictions = predictions.values.tolist()\n",
+    "    print(f\"Model Inference output: \")\n",
+    "    for i, model_output in enumerate(predictions):\n",
+    "        print(f\"Record: {i + 1}\\tModel Prediction: {model_output[0]}\")\n",
+    "\n",
+    "    if \"kernel_shap\" in result[\"explanations\"]:\n",
+    "        explanations = result[\"explanations\"][\"kernel_shap\"]\n",
+    "        display_plots(explanations, expected_value, request_records, predictions)\n",
+    "    else:\n",
+    "        print(f\"No Clarify explanations for the record(s)\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "58320890",
+   "metadata": {},
+   "source": [
+    "## Prepare data\n",
```

Comment:
For this section and the training section, good to mention that the content/code is reused from another notebook https://github.com/aws/amazon-sagemaker-examples/blob/main/advanced_functionality/multi_model_sklearn_home_value/sklearn_multi_model_endpoint_home_value.ipynb